registers/dump/main.go

```diff
@@ -0,0 +1,35 @@
+package main
+
+import (
+	"flag"
+	"log"
+
+	"github.com/benlaurie/gds-registers/register"
+)
+
+var (
+	regname = flag.String("register", "register", "name of register (e.g. 'country')")
+)
+
+type dumper struct {
+}
+
+func (*dumper) Process(e map[string]interface{}, h string, i map[string]interface{}) error {
+	log.Printf("%#v %s %#v", e, h, i)
+	return nil
+}
+
+func main() {
+	flag.Parse()
+
+	r, err := register.NewRegister(*regname)
+	if err != nil {
+		log.Fatal(err)
+	}
+	log.Printf("%#v", r)
+
+	err = r.GetEntries(&dumper{})
```

Comment:
```golang
if err := r.GetEntr...; err != nil {
  ...
}
```

Reply:
I find that less readable, is it a style guide thing, or personal preference?

Reply:
That's the common style; unless the initialisation returns a value you need to have outside the `if` scope, or the func call is too long (e.g. if you're passing in an inline func() {...} as a parameter, then prefer this.

https://golang.org/doc/effective_go.html#if

Reply:
Wot about this one then? :)